ci/Jenkinsfile

```diff
@@ -0,0 +1,120 @@
+    }
+
+    parameters {
+        string(
+          name: 'BRANCH',
+          description: 'Paste a branch name from test repository.',
+          defaultValue: 'master'
+        )
+        string(
+          name: 'BUILD',
+          description: 'Paste a number of build that you want to test.',
+          defaultValue: 'PR-'
+        )
+        string(
+          name: 'FILTER',
+          description: 'Paste "-k" with test name/part of test name to run specific test or/and "-m" with tag to run specific scope of tests',
+          defaultValue: ''
+        )
+
+        string(
+          name: 'TEST_RUN',
+          description: 'Test run ID in Test Rail',
+          defaultValue: ''
+        )
+    }
+
+    options {
+        timestamps()
+        /* Prevent Jenkins jobs from running forever */
+        timeout(time: 120, unit: 'MINUTES')
+        /* manage how many builds we keep */
+        buildDiscarder(logRotator(
+            numToKeepStr: '10',
+            daysToKeepStr: '30',
+            artifactNumToKeepStr: '3',
+        ))
+    }
+
+    stages {
+        stage('Clean') {
+            steps {
+                sh "rm -rf ${env.WORKSPACE}/tmp/"
+                sh "rm -rf ${env.WORKSPACE}/allure-results"
+            }
+        }
+
+        stage('Download') {
+            steps { script {
+                copyArtifacts(
+                    projectName: "status-desktop/branches/linux/x86_64/package/${params.BUILD}/",
+                    filter: 'pkg/*-x86_64.tar.gz',
+                    selector: lastWithArtifacts(),
+                    target: "${env.WORKSPACE}"
+                )
+                pkg_path = "${env.WORKSPACE}/${utils.findFile('pkg/*-x86_64.tar.gz')}"
+                sh "tar -zxvf '${pkg_path}' -C '${env.WORKSPACE}'"
+                app_dir = "${env.WORKSPACE}/${utils.findFile('*.AppImage')}"
+                env.APP_DIR=app_dir
+            } }
+        }
+
+        stage('Setup') {
+            steps { script {
+                sh 'pip3 install --user -r requirements.txt'
+            } }
+        }
+
+        stage('Test') {
+            steps {
+                withCredentials([
+                    usernamePassword(
+                    credentialsId:  'test-rail-api-devops',
+                    usernameVariable: 'TESTRAIL_USER',
+                    passwordVariable: 'TESTRAIL_PWD'
+                )]) { wrap([
+                        $class: 'Xvfb',
+                        autoDisplayName: true,
+                        parallelBuild: false,
+                        screen: '1920x1080x24',
+                    ]) { script {
+                        if (params.TEST_RUN != '') {
+                            env.TESTRAIL_RUN_ID = "${params.TEST_RUN}"
+                            env.TESTRAIL_URL = "https://ethstatus.testrail.net"
+                            env.TESTRAIL_USER = "${TESTRAIL_USER}"
+                            env.TESTRAIL_PWD = "${TESTRAIL_PWD}"
+                        }
```

Comment:
This whole section doesn't need to be here, move it to `environment` section.

Also, why are you naming the parameter `TEST_RUN` and then renaming it `TESTRAIL_RUN_ID`. Just name the parameter `TESTRAIL_RUN_ID` from the start.

Reply:
Fixed